src/main/java/Duke.java

```diff
@@ -1,10 +1,152 @@
+
+        if (description.isEmpty() && by.isEmpty()) {
+            throw new DukeException("Please input the description and deadline of the task!");
+        } else if (description.isEmpty()) {
+            throw new DukeException("Please input the description of the task!");
+        } else if (by.isEmpty()) {
+            throw new DukeException("Please input the deadline of the task!");
+        }
+
+        Deadline newDeadline = new Deadline(description, by);
+        taskManager.addTask(newDeadline);
+    }
+
+    private static void handleEvent(String[] inputSplit) throws DukeException {
+        int indexOfAt = Arrays.asList(inputSplit).indexOf("/at");
+
+        if (indexOfAt == -1) {
+            throw new DukeException("Please input the description and/or duration of the event!");
+        }
+
+        String description = String.join(" ", Arrays.copyOfRange(inputSplit, 1, indexOfAt));
+        String at = String.join(" ",
+                Arrays.copyOfRange(inputSplit, indexOfAt + 1, inputSplit.length));
+
+        if (description.isEmpty() && at.isEmpty()) {
+            throw new DukeException("Please input the description and duration of the event!");
+        } else if (description.isEmpty()) {
+            throw new DukeException("Please input the description of the event!");
+        } else if (at.isEmpty()) {
+            throw new DukeException("Please input the duration of the event!");
+        }
+
+        Event newEvent = new Event(description, at);
+        taskManager.addTask(newEvent);
+    }
+
+    private static void handleDelete(String[] inputSplit) throws DukeException {
+        try {
+            int indexOfTask = Integer.parseInt(inputSplit[1]) - 1;
+            taskManager.deleteTask(indexOfTask);
+        } catch (NumberFormatException e) {
+            throw new DukeException("Please input a valid index of task!");
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Please input the index of task you want to delete!");
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Scanner sc = new Scanner(System.in);
+
+        taskManager.printIntroduction();
+
+        while (true) {
+            String input = sc.nextLine();
+            String[] inputSplit = input.split(" ");
+            String command = input.trim().toLowerCase().split(" ")[0];
+
+            switch (command) {
+            case "list":
+                taskManager.printTasks();
+                break;
+            case "bye":
+                taskManager.printGoodBye();
+                return;
+            case "done":
+                try {
```

Comment:
The try catch loop is similar for several switch cases. Could consider extracting them to a general method called handleInput(inputSplit)?